File: JAVA/2300. Successful Pairs of Spells and Potions.java
class Solution {
    public int[] successfulPairs(int[] s, int[] p, long t) {
        int[] a=new int[s.length];
        Arrays.sort(p);
        for(int i=0;i<s.length;i++){
            int l=0,r=p.length;
            while(l<r){
                int m=l+(r-l)/2;
                if((long)s[i]*p[m]>=t)
                r=m;
                else
                l=m+1;
            }
            a[i]=p.length-l;
        }
        return a;
    }
}

File: C++/2300. Successful Pairs of Spells and Potions.cpp
class Solution {
public:
    vector<int> successfulPairs(vector<int>& s, vector<int>& p, long long t) {
        vector<int>a;
        sort(p.begin(),p.end());
        for(int i=0;i<s.size();i++){
            int l=0,r=p.size();
            while(l<r){
                int m=l+(r-l)/2;
                if((long long)s[i] * p[m] >= t) r=m;
                else l=m+1;
            }
            a.push_back(p.size()-l);
        }
        return a;
    }
};
